utils/src/test/java/io/helidon/build/util/HelidonVersionsTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.build.util;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import static io.helidon.build.util.MavenVersion.greaterThanOrEqualTo;
+import static io.helidon.build.util.MavenVersion.notQualified;
+import static io.helidon.build.util.MavenVersion.toMavenVersion;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * Unit test for class {@link HelidonVersions}.
+ */
+class HelidonVersionsTest {
+    private static final String TWO_0_0_VERSION_STRING = "2.0.0";
+    private static final String HIGH_VERSION_STRING = "99999.99999.99999";
+    private static final String HIGHER_VERSION_STRING = "199999.99999.99999";
+    private static final MavenVersion TWO_0_0_VERSION = toMavenVersion(HIGHER_VERSION_STRING);
+    private static final MavenVersion HIGHER_VERSION = toMavenVersion(HIGHER_VERSION_STRING);
+
+    private static final Predicate<MavenVersion> UNQUALIFIED_2_0_0_MINUMUM =
+        notQualified().and(greaterThanOrEqualTo(TWO_0_0_VERSION_STRING));
+
+    private static final Predicate<MavenVersion> HIGH_UNQUALIFIED_MINUMUM =
+        notQualified().and(greaterThanOrEqualTo(HIGH_VERSION_STRING));
+
+    @Test
+    void testFilteredFallbackIsEmpty() {
```

Comment:
Can we use mockito to avoid the dependency on Maven Central ? This way we also avoid the proxy configuration headache.

Reply:
This obviously works on Wercker; do you expect there to be a proxy issue in your new setup?

Reply:
Testing against an external environment like Maven Central should be done separately, ideally in an integration test. Ideally unit tests can be ran offline as long as the Maven repository is populated.

Surefire forks a new JVM and system properties are not propagated. If behind a proxy then this would fail.

Reply:
I grant you that maybe this would be better as an integration test, but... aren't we going to run those as well? How does this solve the (potential) proxy problem?

Reply:
failsafe and surefire are almost identical, so changing this into an integration test won't solve the proxy issue. 

BTW see the current pre-merge pipeline is stuck on `io.helidon.build.util.HelidonVersionsTest`

Also I'm thinking we have to support the "http_proxy" env variables to support proxy with the native cli. This can actually solve the proxy issue in the unit test since environment variables are propagated to forks.

I.e this means we would be assuming "http_proxy" to be set for the build behind proxy to pass.

Reply:
I've added connect and read timeouts, and converted `HelidonVersionsTest` to `HelidonVersionIT`.

Reply:
So this looks like it is now in your hands to solve the proxy issue for the new build. Or do you just want me to disable this test for now?

Reply:
There are only a couple of way to solve the proxy issue: 
- use JAVA_TOOL_OPTIONS
- configure surefire to not fork
- manually propagate system properties in the configuration (using `<systemPropertyVariables>`)

I'd say just set `<forkCount>0</forkCount>`.